src/Networking/NexusMods.Networking.NexusWebApi/ModUpdateService.cs

```diff
@@ -111,10 +176,36 @@ public void NotifyForUpdates()
     }
 
     /// <inheritdoc />
-    public IObservable<Optional<NexusModsFileMetadata.ReadOnly[]>> GetNewestModPageVersionObservable(NexusModsModPageMetadata.ReadOnly current)
+    public IObservable<Optional<NewerFilesOnModPage>> GetNewestModPageVersionObservable(NexusModsModPageMetadata.ReadOnly current)
     {
         return _newestModOnAnyPageCache.Connect()
             .Transform(kv => kv.Value)
-            .QueryWhenChanged(query => query.Lookup(current.Id));
+            .QueryWhenChanged(query =>
+                {
+                    var files = query.Lookup(current.Id);
+                    return files == null
+                        ? Optional<NewerFilesOnModPage>.None
+                           : new NewerFilesOnModPage(files.Value);
+                }
+            );
     }
 }
+
+/// <summary>
+/// Marks all of the files on a mod page that are newer than the current ones.
+/// </summary>
+/// <param name="Files">
+/// All the files on the mod page that are newer, these are sorted by uploaded date, descending,
+/// meaning that the first file in the array is the newest.
+/// </param>
+public readonly record struct NewerFilesOnModPage(NexusModsFileMetadata.ReadOnly[] Files)
+{
+    /// <summary>
+    /// Returns the newest file on the mod page.
+    /// </summary>
+    /// <remarks>
+    /// Note(sewer): We by definition don't create this struct with empty arrays,
+    /// so the first element is always present. 
+    /// </remarks>
+    public NexusModsFileMetadata.ReadOnly NewestFile() => Files[0];
+}
```

Comment:
I would add Debug assert in the constructor to ensure Files is not empty

Reply:
I don't think it's needed here actually. If the assumption is false, that will trigger an unhandled exception which will propagate to the top level error handler, that being the error dialog.

So regardless of whether there is an assert or not, the user would be notified.